src/lib/fork_job.js

```diff
@@ -0,0 +1,24 @@
+/*
+ * Forks a child_process and send it a job and returns a promise.
+ * Resolves if forked process exits error code 0
+ */
+const fork = require('child_process').fork
+const promise = require('./promises/create')
+
+module.exports = (worker, job) => {
+  const deferred = promise()
+
+  const forked = fork(worker)
+
+  forked.send(job)
+
+  forked.on('exit', (code) => {
+    // we did it!
+    if (code === 0) return deferred.resolve()
+
+    // something went wrong!
+    if (code !== 0) return deferred.reject(code)
```

Comment:
Just return a reject here if the code is non-zero, and return the resolve otherwise (1 less `if`), but it's a minor detail

Reply:
yeah return home early!